apps/web/src/views/GaugesVoting/hooks/useGaugesFilter.ts

```diff
@@ -51,6 +59,84 @@ export const useGaugesFilter = (fullGauges: Gauge[] | undefined) => {
     [filter],
   )
 
+  return {
+    searchText,
+    setSearchText,
+
+    filter,
+    setFilter,
+    onFilterChange,
+  }
+}
+
+const useGaugesFilterQueryState = () => {
+  const [searchText, setSearchText] = useQueryState(
+    'q',
+    parseAsString.withDefault('').withOptions({
+      history: 'replace',
+      shallow: true,
+    }),
+  )
+  const [filter, setFilter] = useQueryStates(
+    {
+      byChain: parseAsArrayOf(parseAsInteger).withDefault([]),
+      byFeeTier: parseAsArrayOf(
+        parseAsNumberLiteral<FeeAmount>([FeeAmount.LOW, FeeAmount.MEDIUM, FeeAmount.HIGH, FeeAmount.LOWEST]),
+      ).withDefault([]),
+      byType: parseAsArrayOf(parseAsStringLiteral<Gauges>([Gauges.Regular, Gauges.Boosted, Gauges.Capped])).withDefault(
+        [],
+      ),
+    },
+    {
+      history: 'replace',
+      shallow: true,
+    },
+  )
+
+  const onFilterChange = useCallback(
+    (type: OptionsType, value: FilterValue) => {
+      const opts = filter[type]
+      // select all
+      if (Array.isArray(value)) {
+        setFilter({
+          [type]: value.length === opts.length ? [] : value,
+        })
+      } else if (opts.some((x) => x === value)) {
+        // deselect one
+        setFilter({
+          [type]: opts.filter((v) => v !== value),
+        })
+      } else {
+        // select one
+        setFilter({
+          [type]: [...opts, value],
+        })
+      }
+    },
+    [filter, setFilter],
+  )
+
+  return {
+    searchText,
+    setSearchText,
+
+    filter,
+    setFilter,
+    onFilterChange,
+  }
+}
+
+const useGaugesFilterState = (useQuery: boolean) => {
+  const pureHook = useGaugesFilterPureState()
+  const queryHook = useGaugesFilterQueryState()
+
+  return useQuery ? queryHook : pureHook
```

Comment:
I think one of the downsides of this implementation is that, we always run an unnecessary hook. Can we use whichever we need depends on the situation?

Reply:
refactor to use two entry hooks, each call different impl